File: programmers/Level2/소수찾기.java
import java.util.*;

class Solution {
    HashSet<Integer> set = new HashSet<>();
    String[] arr; 
    boolean[] visited;
    
    public int solution(String numbers) {
        arr = numbers.split("");
        visited = new boolean[arr.length];

        for (int i = 1; i <= arr.length; i++) {
            permutation(arr.length, i, "", 0);
        }
        
        int answer = 0;
        Iterator iter = set.iterator();
        
        while (iter.hasNext()) {
            int n = (int) iter.next();
            if (isPrime(n)) {
                answer++;
            }
        }
        
        return answer;
    }
    
    public void permutation(int n, int r, String s, int depth) {
        if (depth == r) {
            set.add(Integer.parseInt(s));
            return;
        }
        
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                visited[i] = true;
                permutation(n, r, s + arr[i], depth + 1);
                visited[i] = false;
            }
        }
    }
    
    public boolean isPrime(int n) {
        if (n == 0 || n == 1) {
            return false;
        }
        
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false;
            }
        }
        
        return true;
    }
}

File: programmers/Level2/소수찾기.py
import math
from itertools import permutations

def isPrime(number):
    if number in [0, 1]:
        return False
    
    for i in range(2, int(math.sqrt(number))+1):
        if number % i == 0:
            return False
        
    return True

def solution(numbers):
    answer = []
    
    for i in range(1, len(numbers)+1):
        perm = list(map(''.join, permutations(list(numbers), i)))
        for j in range(len(perm)):
            num = int(perm[j])
            if isPrime(num):
                answer.append(num)

    answer = list(set(answer))
    return len(answer)